mvlearn/semi_supervised/ctregression.py

```diff
@@ -278,8 +276,8 @@ def fit(self, Xs, y):
         L2 = L.copy()
 
         # fitting the estimator object on the train data
-        self.estimator1_.fit(X1[L1], y1[L1])
-        self.estimator2_.fit(X2[L2], y2[L2])
+        self.estimator1.fit(X1[L1], y1[L1])
+        self.estimator2.fit(X2[L2], y2[L2])
```

Comment:
this was correct. See self.estimator1 as a kind of factory that you can copy/clone to get an instance you want to use at fit time. The fit method should only modify parameters that end with _. The self.estimator1 should stay untouched. It means you need both self.estimator1 and self.estimator1_ (the clean one from init you could reuse for a new fit on a new dataset) and the one used at fit one the last data.

Reply:
got it, I had made the change in ctclassifier but not ctregression. Now I make a copy of the original estimator1 and estimator2 each time ``fit`` is called